pre-docs/bolt_project_directory.md

```diff
@@ -0,0 +1,115 @@
+```
+project/
+├── Puppetfile
+├── bolt.yaml
+├── data
+│   └── common.yaml
+├── inventory.yaml
+└── site-modules
+    └── project
+        ├── manifests
+        │   └── my_class.pp
+        ├── plans
+        │   ├── deploy.pp
+        │   └── diagnose.pp
+        └── tasks
+            ├── init.json
+            └── init.py
+```
+
+> **Tip**: You can use an existing control repo as a bolt project directory by
+> adding a `bolt.yaml` file to it and cofiguring the `modulepath` to match the
+> `modulepath` in `enviroment.conf`
+
+### Embedded Project Directory
+
+If the current directory or a parent contains a directory called `Boltdir` Bolt
+will use that as the project directory. Use this type of project directory to
+embed bolt management code into another repo. For example you can store
+management code in the same repo as the application it manages without
+cluttering up the top level with multiple files. This will allow you to run
+bolt from anywhere in the applications directory structure.
+
+An embedded project directory has a structure like this:
+```
+project/
+├── Boltdir
+│   ├── Puppetfile
+│   ├── bolt.yaml
+│   ├── data
+│   │   └── common.yaml
+│   ├── inventory.yaml
+│   └── site-modules
+│       └── project
+│           ├── manifests
+│           │   └── my_class.pp
+│           ├── plans
+│           │   ├── deploy.pp
+│           │   └── diagnose.pp
+│           └── tasks
+│               ├── init.json
+│               └── init.py
+├── src <non bolt source code for the project>
+└── tests <non bolt tests for the project>
+```
+
+**note** If a directory contains both `Boltdir` and `bolt.yaml` the `Boltdir`
+directory will be used as the project directory rather then the parent.
+
+
+### User Project Directory
+
+If bolt cannot find a Project directory based on `Boltdir` or `bolt.yaml` it
+will use `~/.puppetlabs/bolt` as the project directory. Use this type of
+project directory if  you have a single set of bolt code and data you use
+across all projects.
+
+### How the project directory is chosen.
+
+The following methods are used by bolt in priority order to chose a project directory.
+
+#. Manually specified: You can specify what directory bolt should use with
+   `--boltdir <directory path>` on the command line. There is not an equivalent
+   configuration setting since the bolt project directory needs to be known to
+   load configuration.
+
+#. Parent directory: Bolt will traverse parents of the current directory until
+   if finds a directory containing a `Boltdir` or `bolt.yaml` or it reaches the
+   root of the file system.
+
+#. User Project Directory: If no directory is specified manually or found in a
+   parent the user project directory will be used.
+
+## Stucture of the project directory
+
+The default paths for all of bolt's configuration, code and data are relative to the modulepath.
+
+* (`bolt.yaml`)[./bolt_configuration_options.md] contains configuration options for bolt.
```

Comment:
Not quire sure why none of these relative links seem to be rendering. Could be the back ticks or maybe relative links to same directory don't respect the `./`. 
![image](https://user-images.githubusercontent.com/15354214/53057333-e3d05a00-3463-11e9-8462-be45deef19ee.png)